webapp/components/features/FollowList/FollowList.vue

```diff
@@ -0,0 +1,157 @@
+    </template>
+    <template v-if="this.connections && this.connections.length <= 7">
+      <ds-space v-for="follow in uniq(this.connections)" :key="follow.id" margin="x-small">
+        <!-- TODO: find better solution for rendering errors -->
+        <client-only>
+          <user-teaser :user="follow" />
+        </client-only>
+      </ds-space>
+      <ds-space v-if="this.allConnectionsCount - this.connections.length" margin="small">
+        <base-button
+          @click="fetchConnections"
+          :loading="this.isLoading"
+          size="small"
+          color="softer"
+        >
+          {{
+            $t('profile.network.andMore', {
+              number: this.allConnectionsCount - this.connections.length,
+            })
+          }}
+        </base-button>
+      </ds-space>
+    </template>
+    <template v-else-if="this.connections.length > 7">
+      <div class="overflow-container">
+        <ds-space
+          v-for="follow in uniq(this.filteredConnections)"
+          :key="follow.id"
+          margin="x-small"
+        >
+          <client-only>
+            <user-teaser :user="follow" />
+          </client-only>
+        </ds-space>
+      </div>
+      <ds-space margin="x-small">
+        <ds-input
+          @input.native="setFilter"
+          :placeholder="filter"
+          v-focus="true"
+          size="small"
+          icon="filter"
+        />
+      </ds-space>
+    </template>
+  </base-card>
+</template>
+
+<script>
+import uniqBy from 'lodash/uniqBy'
+import UserTeaser from '~/components/UserTeaser/UserTeaser'
+import { followedByQuery, followingQuery } from '~/graphql/User'
+
+export default {
+  name: 'FollowerList',
+  components: {
+    UserTeaser,
+  },
+  props: {
+    user: { type: Object, default: null },
+    type: { type: String, default: 'following' },
+  },
+  data() {
+    return {
+      additionalConnections: [],
+      filter: null,
+      isLoading: false,
+      queries: {
+        followedBy: followedByQuery,
+        following: followingQuery,
+      },
+    }
+  },
+  computed: {
+    userName() {
+      const { name } = this.user || {}
+      return name || this.$t('profile.userAnonym')
+    },
+    connections() {
+      return [...this.user[this.type], ...this.additionalConnections]
+    },
+    filteredConnections() {
+      if (!this.filter) {
+        return this.connections
+      }
+
+      const fuzzyExpression = new RegExp(
+        `${this.filter.split('').reduce((part, c) => `${part}[^${c}]*${c}`)}`,
+        'i',
+      )
+
+      const fuzzyScores = this.connections
+        .map((user) => {
+          const match = user.name.match(fuzzyExpression)
+
+          return {
+            user,
+            score: match ? match[0].length * (match.index + 1) : -1,
+          }
+        })
+        .filter((score) => score.score !== -1)
+        .sort((a, b) => a.score - b.score)
+
+      return fuzzyScores.map((score) => score.user)
```

Comment:
this looks complex, I'll have to sit down and think about what this does, and if there's not a way to simplify it

Reply:
Second that. I'll skip this bit for now – if you want to have a look together, let me know @mattwr18 😅

Reply:
I like it and copied it to my knowledge base :)

Reply:
Thanks, @Mogge 
It fails to prioritize full matches when the strings are longer (>15). In some cases, the first match is long enough to overtake a 1-2 separation match further to the start.
```js
// works as snippet in browser dev-tools
(() => {
  const fuzzyFilter = (str, filter) => {
    const fuzzyExpression = new RegExp(
      `${filter.split('').reduce((expr, c) => `${expr}([^${escape(c)}]*${escape(c)})`, '')}`,
      'i',
    )
    const match = str.match(fuzzyExpression)
    if (!match) {
      return false
    }
    let score = 1
    for (let i = 1; i <= filter.length; i++) {
      score *= match[i].length
    }
    return {
      str,
      filter,
      fuzzyExpression,
      match,
      score,
    }
  }
  // .score === 13
  console.log(fuzzyFilter('thank you, very much', 'ery'))
  // .score === 12 (i.e. higher in the list)
  console.log(fuzzyFilter('Pecurly', 'ery'))
})();